python/paddle/fluid/contrib/mixed_precision/bf16/amp_utils.py

```diff
@@ -191,7 +232,180 @@ def bf16_guard():
         yield
 
 
-def rewrite_program_bf16(main_prog, amp_lists=None, use_bf16_guard=False):
+def cast_model_to_bf16(program, amp_lists=None, use_bf16_guard=True):
+    """
+    Traverse all ops in the whole model and set their inputs and outputs
+    to the bf16 data type. This function will do some special process for
+    the batch normalization, which keeps the computational process of
+    batchnorms in FP32.
```

Comment:
```suggestion
    to the bf16 data type. This function will do some special processing for
    the batch normalization, which will keep the batchnorm's computations
    in FP32.
```
Additionally why batchnorm is not supported/specially treated? Shouldn't it be in unsupported list then?

Reply:
BatchNorm op can be run in bf16 if needed (in accuracy/speed experiments ), however the exception in the code is about X, Z inputs not to be forced to bf16 in pure mode. Other inputs don't have that restriction. Unsupported list is intended to be list of ops not implemented for bf16 datatype (not present in the code), so not being able to be called.